examples/chatbot_llm/main.py

```diff
@@ -1,25 +1,152 @@
+                                  context length of the pipeline.  [default:
+                                  512]
+  --sampling_temperature FLOAT    The temperature to use when samplingfrom the
+                                  probability distribution computed from the
+                                  logits.Higher values will result in more
+                                  random samples. Shouldbe greater than 0.0.
+                                  [default: 1.0]
+  --prompt_length INTEGER         Processed prompt in chunks of this length.
+                                  This is to maximize the inference speed
+                                  [default: 64]
+  --token_generation_speed / --no-token_generation_speed
+                                  Whether to display the token generation
+                                  speed or not  [default: no-
+                                  token_generation_speed]
+  --help                          Show this message and exit.
+
+Examples:
+
+1) Use a local deployment directory
+python main.py models/llama/deployment
+
+2) Use a SparseZoo stub
+python main.py \
+    zoo:nlg/text_generation/codegen_mono-350m/pytorch/huggingface/bigpython_bigquery_thepile/base-none # noqa: E501
+
+3) Display token generation speed
+python main.py models/llama/deployment \
+    --token_generation_speed
+"""
+import click
+
 from deepsparse import Pipeline
 
-def run(args):
 
+@click.command(
+    context_settings=dict(
+        token_normalize_func=lambda x: x.replace("-", "_"), show_default=True
+    )
+)
+@click.argument("model_path", type=str)
+@click.option(
+    "--session_id",
+    "-s",
+    type=str,
+    default="session_1",
+    help="Name of the session that will be used in the example",
+)
+@click.option(
+    "--max_generated_tokens",
+    "-m",
+    type=int,
+    default=32,
+    help="Max number of tokens to generate",
+)
+@click.option(
+    "--sequence_length",
+    type=int,
+    default=512,
+    help="Sequence length to compile model and tokenizer for."
+    "This controls the maximum context length of the pipeline.",
+)
+@click.option(
+    "--sampling_temperature",
+    type=float,
+    default=1.0,
+    help="The temperature to use when sampling"
+    "from the probability distribution computed from the logits."
+    "Higher values will result in more random samples. Should"
+    "be greater than 0.0.",
+)
+@click.option(
+    "--prompt_length",
+    type=int,
+    default=64,
+    help="Processed prompt in chunks of this length. "
+    "This is to maximize the inference speed",
+)
+@click.option(
+    "--token_generation_speed/--no-token_generation_speed",
+    default=False,
+    help="Whether to display the token generation speed or not",
+)
+def main(
+    model_path,
+    session_id,
+    max_generated_tokens,
+    sequence_length,
+    sampling_temperature,
+    prompt_length,
+    token_generation_speed,
+):
```

Comment:
Could we add types and docstrings for these args?

Reply:
Addes types, however the docstring of this function is used as the help text in click, the click decorators do have explanation on what the respective args are.